tests/install_tests.rs

```diff
@@ -356,3 +357,44 @@ async fn install_conda_meta_history() {
 
     assert!(conda_meta_history_file.exists());
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+#[serial]
+#[cfg_attr(not(feature = "slow_integration_tests"), ignore)]
+async fn minimal_lockfile_update_pypi() {
+    let pixi = PixiControl::new().unwrap();
+    pixi.init().await.unwrap();
+
+    // Add and update lockfile with this version of python
+    pixi.add("python==3.11").with_install(true).await.unwrap();
+
+    // Add pypi dependencies which are not the latest options
+    pixi.add_multiple(vec!["uvicorn==0.28.0", "click==7.1.2"])
+        .set_type(pixi::DependencyType::PypiDependency)
+        .with_install(true)
+        .await
+        .unwrap();
+
+    // Check the locked click dependencies
+    let lock = pixi.lock_file().await.unwrap();
+    assert!(lock.contains_pep508_requirement(
+        DEFAULT_ENVIRONMENT_NAME,
+        Platform::current(),
+        pep508_rs::Requirement::from_str("click==7.1.2").unwrap()
+    ));
+
+    // Widening the click version to allow for the latest version
+    pixi.add_multiple(vec!["uvicorn==0.29.0", "click"])
+        .set_type(pixi::DependencyType::PypiDependency)
+        .with_install(true)
+        .await
+        .unwrap();
+
+    // Check the locked click dependencies to see if it was only minimally updated
+    let lock = pixi.lock_file().await.unwrap();
+    assert!(lock.contains_pep508_requirement(
+        DEFAULT_ENVIRONMENT_NAME,
+        Platform::current(),
+        pep508_rs::Requirement::from_str("click==7.1.2").unwrap()
```

Comment:
```suggestion
        Platform::current(),
        // With a fresh solve this would be bumped to `>=8.0.0`
        pep508_rs::Requirement::from_str("click==7.1.2").unwrap()
```